src/server/game/Maps/Map.cpp

```diff
@@ -2989,20 +2990,14 @@ bool InstanceMap::CanEnter(Player* player)
                 if (iPlayer->IsGameMaster()) // bypass GMs
                     continue;
                 if (!player->GetGroup()) // player has not group and there is someone inside, deny entry
-                {
-                    player->SendTransferAborted(GetId(), TRANSFER_ABORT_MAX_PLAYERS);
-                    return false;
-                }
+                    return CANNOT_ENTER_MAX_PLAYERS;
                 // player inside instance has no group or his groups is different to entering player's one, deny entry
                 if (!iPlayer->GetGroup() || iPlayer->GetGroup() != player->GetGroup())
```

Comment:
Generated code should be the same, and your version is harder to read. Also you missed a !, it should be
`!player->GetGroup() || !iPlayer->GetGroup() || player->GetGroup != iPlayer->GetGroup()`
and you could even omit the second check out of the three completely.